CMakeLists.txt

```
@@ -302,22 +302,9 @@ if(OpenMP_FOUND)
   # so we use always OpenMP_CXX_SPEC_DATE, which is available since CMake 3.7.
   message(STATUS "Found OpenMP, spec date ${OpenMP_CXX_SPEC_DATE}")
 
-  if(MSVC)
-    if(MSVC_VERSION EQUAL 1900)
-      set(OPENMP_DLL VCOMP140)
-    elseif(MSVC_VERSION MATCHES "^191[0-9]$")
-      set(OPENMP_DLL VCOMP140)
-    elseif(MSVC_VERSION MATCHES "^192[0-9]$")
-      set(OPENMP_DLL VCOMP140)
-    elseif(MSVC_VERSION MATCHES "^193[0-9]$")
-      set(OPENMP_DLL VCOMP140)
-    endif()
-    if(OPENMP_DLL)
-      string(APPEND CMAKE_SHARED_LINKER_FLAGS_DEBUG " /DELAYLOAD:${OPENMP_DLL}D.dll")
-      string(APPEND CMAKE_SHARED_LINKER_FLAGS_RELEASE " /DELAYLOAD:${OPENMP_DLL}.dll")
-    else()
-      message(WARNING "Delay loading flag for OpenMP DLL is invalid.")
-    endif()
+  if(MSVC AND ((MSVC_VERSION EQUAL 1900) OR (MSVC_VERSION MATCHES "^191[0-9]$")))
```

I think the check for MSVC is not necessary as `MSVC_VERSION` will not contain a valid version number in case it is another compiler ;-)
```suggestion
  if((MSVC_VERSION EQUAL 1900) OR (MSVC_VERSION MATCHES "^191[0-9]$")))
```